spec/models/user_spec.rb

```diff
@@ -0,0 +1,59 @@
+# == Schema Information
+#
+# Table name: users
+#
+#  id                     :integer          not null, primary key
+#  provider               :string           default("email"), not null
+#  uid                    :string           default(""), not null
+#  encrypted_password     :string           default(""), not null
+#  reset_password_token   :string
+#  reset_password_sent_at :datetime
+#  allow_password_change  :boolean          default(FALSE)
+#  remember_created_at    :datetime
+#  sign_in_count          :integer          default(0), not null
+#  current_sign_in_at     :datetime
+#  last_sign_in_at        :datetime
+#  current_sign_in_ip     :string
+#  last_sign_in_ip        :string
+#  confirmation_token     :string
+#  confirmed_at           :datetime
+#  confirmation_sent_at   :datetime
+#  unconfirmed_email      :string
+#  name                   :string
+#  email                  :string
+#  tokens                 :json
+#  created_at             :datetime         not null
+#  updated_at             :datetime         not null
+#  gender                 :integer
+#
+# Indexes
+#
+#  index_users_on_confirmation_token    (confirmation_token) UNIQUE
+#  index_users_on_email                 (email) UNIQUE
+#  index_users_on_reset_password_token  (reset_password_token) UNIQUE
+#  index_users_on_uid_and_provider      (uid,provider) UNIQUE
+#
+
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  let(:email)                 { 'test@test.com' }
+  let(:password)              { '12345678' }
+  let(:name)                  { 'Sebastian' }
+  let(:gender)                { 'abc' }
+
+  it 'is not valid without a name' do
+    user = User.new(email: email, password: password)
+    expect(user).to_not be_valid
+  end
+
+  it 'is not valid without a gender' do
+    user = User.new(email: email, password: password, name: name)
+    expect(user).to_not be_valid
+  end
+
+  it 'is not valid without a valid gender' do
+    expect { User.new(email: email, password: password, name: name, gender: gender) }
+      .to raise_error(ArgumentError)
+  end
```

Comment:
Check ```validate_presence_of```in Rspec.  These tests could be collapsed into 3 lines.
Something like:
 ```subject { build :item }
    it { should validate_presence_of(:name) }```

Reply:
I agree with Gabriel. Also, use expect instead of should (`is_expected.to`)